src/net.rs

```diff
@@ -879,22 +886,20 @@ impl Actor {
                         continue;
                     };
                     let TopicState {
-                        joined,
                         neighbors,
                         event_senders,
                         ..
                     } = state;
-                    let event = if let ProtoEvent::NeighborUp(neighbor) = event {
-                        neighbors.insert(neighbor);
-                        if !*joined {
-                            *joined = true;
-                            GossipEvent::Joined(vec![neighbor])
-                        } else {
-                            GossipEvent::NeighborUp(neighbor)
+                    match &event {
+                        ProtoEvent::NeighborUp(neighbor) => {
+                            neighbors.insert(*neighbor);
                         }
-                    } else {
-                        event.into()
-                    };
+                        ProtoEvent::NeighborDown(neighbor) => {
+                            neighbors.remove(neighbor);
```

Comment:
what's uo with this new line? was this a bug? I'm having trouble following where else is this used since there are like 3 different layers of event with a similar variant

Reply:
Yeah, I think there was a bug previously. Leaving neighbors weren't removed from the neighbor list tracked in `TopicState`. Thus, when setting up a new subscription on `Gossip` (or over RPC) the `Joined` event could include neighbors that actually already left again.